modules/build_dataset.py

```diff
@@ -36,25 +41,76 @@
 
 Franchise_Dict = dict(zip(FRANCHISES, FRANCHISE_NAMES))
 
-start_year = 2009
-end_year = 2021
 
+def get_filename(start_year, end_year):
+    return f'nfl_master_{start_year}-{end_year}.csv'
+
+
+def backup_existing_master(filename):
+    oldFile, newFile = filename, 'backup_' + filename
+    print(f'Backing up\nFrom: {oldFile}\nTo: {newFile}')
+    copyfile(oldFile, newFile)
+
+
+def restore_backup(filename):
+    oldFile, newFile = 'backup_' + filename, filename
+    print(f'Restoring\nFrom: {oldFile}\nTo: {newFile}')
+    copyfile(oldFile, newFile)
+
+
+def parse(f, team, verbose_name, year, week=None):
+    for season in parse_season(team, verbose_name, year, week):
+        print(f'{verbose_name} - {year}')
+        f.write(season + '\n')
+
+
+# recreate master spreadsheet
+def build_master(start_year, end_year):
+    filename = get_filename(start_year, end_year)
+
+    backup_existing_master(filename)
 
-def build_master():
-    filename = 'nfl_master_%s-%s.csv' % (start_year, end_year)
     with open(filename, 'w') as f:
         f.write(','.join(COL_NAMES) + '\n')
         for team, verbose_name in Franchise_Dict.items():
             for year in range(start_year, end_year + 1):
                 try:
-                    for season in parse_season(team, verbose_name, year):
-                        print('{} - {}'.format(verbose_name, year))
-                        f.write(season + '\n')
+                    parse(f, team, verbose_name, year)
                 except Exception as e:
-                    print('Exception!')
-                    print(e)
-                    pass
+                    raise(e)
+
+
+# appends new week to master spreadsheet
+def add_new_week(year, week):
+    filename = get_filename(start_year, end_year)
+
+    backup_existing_master(filename)
+
+    with open(filename, 'a') as f:
```

Comment:
Try to avoid using f as a variable name as its actually a builtin for string formatting

Reply:
Ahh good catch!

Reply:
no biggy just giving a tip for your day job